src/TurbulenceClosures/turbulence_closure_implementations/tke_based_vertical_diffusivity.jl

```diff
@@ -0,0 +1,370 @@
+using Oceananigans.Architectures: architecture, device_event
+using Oceananigans.AbstractOperations: KernelFunctionOperation
+using Oceananigans.BuoyancyModels: ∂z_b
+using Oceananigans.Operators: ℑzᵃᵃᶜ
+
+struct TKEBasedVerticalDiffusivity{TD, CK, CD, CL, CQ} <: AbstractTurbulenceClosure{TD}
+    diffusivity_scaling :: CK
+    dissipation_parameter :: CD
+    mixing_length_parameter :: CL
+    surface_model :: CQ
+
+    function TKEBasedVerticalDiffusivity{TD}(
+        diffusivity_scaling :: CK,
+        dissipation_parameter :: CD,
+        mixing_length_parameter :: CL,
+        surface_model :: CQ) where {TD, CK, CD, CL, CQ}
+
+        return new{TD, CK, CD, CL, CQ}(diffusivity_scaling, dissipation_parameter, mixing_length_parameter, surface_model)
+    end
+end
+
+"""
+    TKEBasedVerticalDiffusivity <: AbstractTurbulenceClosure{TD}
+
+Parameters for the "anisotropic minimum dissipation" turbulence closure for large eddy simulation
+proposed originally by [Rozema15](@cite) and [Abkar16](@cite), and then modified
+by [Verstappen18](@cite), and finally described and validated for by [Vreugdenhil18](@cite).
+"""
+function TKEBasedVerticalDiffusivity(FT=Float64;
+                                     diffusivity_scaling = RiDependentDiffusivityScaling{FT}(),
+                                     dissipation_parameter = 2.91,
+                                     mixing_length_parameter = 1.16,
```

Comment:
Might be useful to document where these numbers are coming from.

Reply:
Ah, they come from a slack message with @adelinehillier ...

Hmm, perhaps I can just include a brief description about the calibration method? I hope we get to update these in the future and, eventually, provide a reference to a paper draft...

Reply:
Ah until then we could just mention it's from an optimization procedure and link to the appropriate repository/script? Just to provide the reader with some context of where the numbers come from.

Reply:
@adelinehillier if there's a script / repo we could link to let us know!

I don't think it's a good idea to spend time creating a repo / context for readers if something isn't readily available. If we don't feel comfortable with these numbers, we could change them all to 1 and say "it's up to the user to choose good parameters".

Reply:
If there's no convenient script/repo then leaving a comment for context is still very helpful to the reader.

Leaving no comment may give the impression that the magic numbers are authoritative in some sense (perhaps from the literature).

Reply:
Ok, I put in a comment about the calibration and included a link to LESbrary.jl. I've also added a warning when building the closure that the parameters are basically arbitrary.

We could also print all the parameters when the closure is built to emphasize this?